src/compile_time_environment.cpp

```diff
@@ -2,8 +2,34 @@
 
 #include "typed_ast.hpp"
 
+#include <cassert>
+
 namespace Frontend {
 
+Binding::Binding(TypedAST::Declaration* decl)
+    : m_type { BindingType::Declaration }, m_decl { decl } {}
+
+Binding::Binding(TypedAST::FunctionLiteral* func, int arg_index)
+    : m_type { BindingType::Argument }
+    , m_func { func }
+    , m_arg_index { arg_index } {}
+
+TypedAST::Declaration* Binding::get_decl() {
+	if(m_type != BindingType::Declaration){ *(int*)0 = 0; }
+	assert(m_type == BindingType::Declaration);
+	return m_decl;
```

Comment:
What is this???

Reply:
Triggering the address sanitizer to get a stack trace

Reply:
Clever, please comment `// address sanitizer` or something next time

Reply:
It's debug code, it won't be there one the final pr